key/src/main/java/software/amazon/kms/key/BaseHandlerStd.java

```diff
@@ -156,4 +158,23 @@ protected void resourceStateCheck(final KeyMetadata keyMetadata) {
           return ProgressEvent.defaultFailureHandler(e, HandlerErrorCode.NotFound);
       return ProgressEvent.defaultFailureHandler(e, HandlerErrorCode.GeneralServiceException);
     }
+
+    /**
+     * A helper method for validating that the requested resource model transition is possible
+     */
+    protected static ProgressEvent<ResourceModel, CallbackContext> validateResourceModel(
+            final ProgressEvent<ResourceModel, CallbackContext> progress, final ResourceModel previousModel,
+            final ResourceModel model) {
+        // If key is not being updated and is disabled, then we cannot perform any updates on the key
+        if (previousModel != null && !previousModel.getEnabled() && !model.getEnabled() &&
+                previousModel.getEnableKeyRotation() != model.getEnableKeyRotation())
+            throw new CfnInvalidRequestException("You cannot modify the EnableKeyRotation property when the Enabled property is false. Set Enabled to true to modify the EnableKeyRotation property.");
+
+        // If the key is asymmetric, we cannot enable key rotation
+        if (!model.getKeySpec().equals(CustomerMasterKeySpec.SYMMETRIC_DEFAULT.toString()) &&
+                model.getEnableKeyRotation())
```

Comment:
what if it is specified as `false`? should we reject the request as well? by default the model adapter sets it to false if not provided - maybe we should keep it as null if asymmetric key?

Reply:
I do not think we should reject the request if this is set to false for an asymmetric key. Asymmetric keys do not have key rotation enabled. I do not see a compelling reason to take away the customer's ability to specify this explicitly in their CloudFormation resource.

Reply:
what is the behavior? will it throw an exception or simply go through as it's false by default?

Reply:
It will work since it is false by default, no exceptions will be thrown